### DISCUSSION TITLE
Action doesn't understand the 'secrets' syntax copied from the documentation

### DISCUSSION THREAD

**Discussion by artis3n** [community/community#27054] [Actions]
<p>I’m trying to pass in a secrets variable as input to my Github Action and I am getting the following error:</p>
<p>“##[error](Line: 7, Col: 18): Unrecognized named-value: ‘secrets’. Located at position 1 within expression: secrets.GALAXY_API_KEY”</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/30ba24c760206bcd61757bf93d2dd1e5c8f2b661.png" data-download-href="/uploads/short-url/6X3Jeo1w5fUtHHK8dMZObfLKwTL.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6969296/181101496-314cc9df-3f3e-40c2-9551-d89b2ca84b20.png" alt="image.png" data-base62-sha1="6X3Jeo1w5fUtHHK8dMZObfLKwTL" width="690" height="203" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/30ba24c760206bcd61757bf93d2dd1e5c8f2b661_2_690x203.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/30ba24c760206bcd61757bf93d2dd1e5c8f2b661_2_1035x304.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/30ba24c760206bcd61757bf93d2dd1e5c8f2b661.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/30ba24c760206bcd61757bf93d2dd1e5c8f2b661_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1067×315 92.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’m following <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/virtual-environments-for-github-actions#creating-and-using-secrets-encrypted-variables" rel="nofollow noopener">this documentation</a>, which has an example showing usage of secrets the same way I am.</p>
<p>My usage: <a href="https://github.com/artis3n/github_version-ansible_plugin/blob/master/.github/workflows/ansiblegalaxy.yml#L32" rel="nofollow noopener">https://github.com/artis3n/github_version-ansible_plugin/blob/master/.github/workflows/ansiblegalaxy.yml#L32</a></p>
<p>Example failed job: <a href="https://github.com/artis3n/github_version-ansible_plugin/commit/f05d3d777beea879d1fa14d38c416b548635ac35/checks?check_suite_id=276794137" rel="nofollow noopener">https://github.com/artis3n/github_version-ansible_plugin/commit/f05d3d777beea879d1fa14d38c416b548635ac35/checks?check_suite_id=276794137</a></p>
<p>And my secrets panel:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6ae96ffe03db84b74323551397420c32de41039b.png" data-download-href="/uploads/short-url/ffMKssFo1NJp58Ae1Vf8DVQHk39.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6969296/181101497-8ac6788d-91fa-459c-b121-ba985168e09a.png" alt="image.png" data-base62-sha1="ffMKssFo1NJp58Ae1Vf8DVQHk39" width="690" height="244" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/6ae96ffe03db84b74323551397420c32de41039b_2_690x244.png, https://aws1.discourse-cdn.com/github/original/2X/6/6ae96ffe03db84b74323551397420c32de41039b.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/6/6ae96ffe03db84b74323551397420c32de41039b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6ae96ffe03db84b74323551397420c32de41039b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">755×268 31.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by artis3n** [comment, marked as answer]
<p>I’m pretty sure this is the problem: <a href="https://github.com/artis3n/ansible_galaxy_collection/pull/11/files#diff-689e1a63f106e616debbed3b5aedcb43L8" rel="nofollow noopener">https://github.com/artis3n/ansible_galaxy_collection/pull/11/files#diff-689e1a63f106e616debbed3b5aedcb43L8</a></p>
<p>I am guessing Github Actions tries to interpolate the value in the description string instead of leaving it raw.</p>
<p>EDIT: Yeah, this was the problem. This seems like a bug with Github Actions. Or is there a way to reference variables and telling Actions not to interpolate them? Some option to use a literal string?</p>

**Comment by molteninjabob** [comment]
<p><a class="mention" href="/artis3n">@artis3n</a> I’m setting up a workflow template to be used by multiple repos and running into this same issue. In the workflow file deployed to each service, it is pulling in the template repository to run scripts as steps. Using secrets directly in the workflow pulled in the secrets just fine, but trying to reference them from another repository is giving me the error. Were you able to find a solution for this or did you just remove the reference?</p>

**Comment by artis3n** [comment]
<p>My issue was <code>{{ }}</code> in my <code>action.yml</code> was getting interpolated instead of being a raw string. I ended up removing the reference.</p>

**Comment by radhikari-arch** [comment]
<p><a class="mention" href="/artis3n">@artis3n</a> I still don’t get it how you resolved. What reference do you mean? I was trying to use the following in the action.</p>
<pre><code class="lang-auto">env:
  MY_SECRET: "${{ secrets.MY_SECRET }}"
</code></pre>
<p>Still getting the error: Unrecognized named-value: ‘secrets’.</p>

**Comment by artis3n** [comment]
<p>My issue was that I was using the secrets interpolation syntax in a file unrelated to the actions workflow - but GitHub was attempting to interpolate that file anyway. See <a href="https://github.com/artis3n/ansible_galaxy_collection/pull/11/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6L8" class="inline-onebox" rel="noopener nofollow ugc">Fix list on README by artis3n · Pull Request #11 · artis3n/ansible_galaxy_collection · GitHub</a></p>
<p>If you are seeing this, double check whether you have ${{ secrets.* }} anywhere else in your repo or PR. The error might not be inside the action workflow file.</p>

**Comment by radhikari-arch** [comment]
<p>Thanks for the response. I find that actions.yml cannot fetch the secrets directly. So, I finally fixed it using with in workflow to pass the secret as:</p>
<pre><code class="lang-auto">with:
  secret_input: ${{ secrets.SUPER_SECRET }}
</code></pre>
<p>And then, in action.yml,</p>
<pre><code class="lang-auto">inputs:
  secret_input: 
     description: "..."
     required: true
...

runs:
.....
  - name: .....
    env:
      MY_SECRET: ${{ inputs.secret_input }}
</code></pre>
<p>This helped: <a href="https://github.com/launchdarkly/find-code-references" class="inline-onebox" rel="noopener nofollow ugc">GitHub - launchdarkly/find-code-references: Used by the GitHub Marketplace</a></p>

**Reply by duvalbruno** [reply]
might be late, but this could help https://docs.github.com/en/actions/security-guides/automatic-token-authentication. Check the part where the doc mention github.token.

**Reply by GregYeo** [reply]
Why are there so many thumbs down reaction? It is nice answer including the source of the problem.

**Reply by sean-m-sullivan** [reply]
@GregYeo it doesn't preserve the secret as a secret, just makes it public, defeating the point of a secret.
